tests/integration/steps/integration_steps.go

```diff
@@ -285,19 +345,46 @@ func CheckSecretIsPresent(t *testing.T, namespace, secret string) {
 
 func CheckNamespaceIsAbsent(t *testing.T, namespace string) {
 	g(t).Eventually(func() bool {
-		return namespaceIsAbsent(t, namespace) || namespaceIsTerminating(t, namespace)
+		absent, err := namespaceIsAbsent(namespace)
+		if err == nil {
+			if absent {
+				return true
+			}
+		} else {
+			t.Logf("failed to check if namespace %s is absent: %s", namespace, err)
+		}
+
+		terminating, err := namespaceIsTerminating(namespace)
+		if err == nil {
+			if terminating {
+				return true
+			}
+		} else {
+			t.Logf("failed to check if namespace %s is terminating: %s", namespace, err)
+		}
+
+		return false
 	}, retryTimeout, retryInterval).Should(BeTrue())
 }
 
-func namespaceIsAbsent(t *testing.T, namespace string) bool {
-	namespaceObject, _ := k8s.GetNamespaceE(t, getKubectlOptions("default"), namespace)
-	return namespaceObject.Name != namespace
+func namespaceIsAbsent(namespace string) (bool, error) {
+	if _, err := GetNamespace(namespace); err == nil {
+		return false, nil
+	} else {
+		if apierrors.IsNotFound(err) {
+			return true, nil
+		} else {
+			return false, err
```

Comment:
Wouldn't it make the code less verbose if we used an assertion to fail the test here in case `err != nil` ?
It would remove a conditional branch in the functions that call `namespaceIsAbsent()`.

Reply:
I am ok with the verbosity here. It makes the function a more general purpose, helper function.

Reply:
Fair enough.

Reply:
Nitpick though, there's two extra else clauses in this function that are not go-ish (most static analyzers should complain) ;)